### PR TITLE
Feature - admin delete resources

### PR DESCRIPTION
**Pull request by alrik**
This PR adds support for deleting users, channels and subscriptions via admin api.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Introduce new administrative channel management endpoints `/admin/v1/realms/{realmId}/channels`.
 - Introduce new administrative subscription management endpoints `/admin/v1/realms/{realmId}/subscriptions`.
 - Introduce new administrative endpoint for deleting auth tokens `DELETE /admin/v1/realms/{realmId}/tokens/{tokenId}`
+- Introduce new administrative endpoint for deleting users `DELETE /admin/v1/realms/{realmId}/users/{userId}`
```
Comment (baranga) [inline CHANGELOG.md:12]: **issue:** The given changelog entry does not cover all new delete endpoints.
**suggestion:** Extend this list to cover all new delete endpoints.

```suggestion
- Introduce new administrative endpoint for deleting channels `DELETE /admin/v1/realms/{realmId}/channels/{channelId}`.
- Introduce new administrative endpoint for deleting subscriptions `DELETE /admin/v1/realms/{realmId}/subscriptions/{subscriptionsId}`.
- Introduce new administrative endpoint for deleting users `DELETE /admin/v1/realms/{realmId}/users/{userId}`.
```

File: src/tasks/admin/delete-subscription.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const subscription = await subscriptionRepository.findOne({realmId, id});
+    if (!subscription) {
+      return failure(taskError({
+        code: 'UnknownSubscription',
+        message: 'Subscription does not exists.',
+      }));
+    }
+
+    await subscriptionRepository.deleteOne({realmId, id});
+    sendSubscriptionSyncMessage({subscription, action: 'deleted'});
```
Comment (baranga) [inline src/tasks/admin/delete-subscription.js:41]: **thought:** Existing subscriptions within the broker are not covered by this. As long as a client holds a connection with an existing subscription open the client is able to receive messages on that channel.

File: src/tasks/admin/delete-user.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        message: 'User does not exists.',
+      }));
+    }
+
+    await Promise.all([
+      userRepository.deleteOne({realmId, id}),
+      authRepository.deleteAllByUserId({realmId, userId: id}),
+      subscriptionRepository.deleteAllByUserId({realmId, userId: id}),
+      realtimeConnectionRepository.deleteAllByUserId({realmId, userId: id}),
+    ]);
```
Comment (baranga) [inline src/tasks/admin/delete-user.js:50]: **thought:** Similar to above :point_up:  this does not closes existing broker connections. Connected clients won't be kicked out and still be able to receive messages on all channels they previously subscribed.

Comment (baranga) [reply src/tasks/admin/delete-user.js:50]: VerneMQ allows to [kick users via command](https://docs.vernemq.com/live-administration/managing-sessions#managing-sessions). Something similar should be possible to be implemented as a broker side API endpoint or some kind message handler within `$SYS`.

File: src/tasks/admin/delete-subscription.js (hunk 2 above)
Comment (baranga) [reply src/tasks/admin/delete-subscription.js:41]: VerneMQ allows to [reevaluate existing subscriptions via command](https://docs.vernemq.com/live-administration/managing-sessions#managing-sessions). Something similar should be possible to be implemented as a broker side API endpoint or some kind message handler within `$SYS`.

**Review by baranga — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (alrik) [reply CHANGELOG.md:12]: The changelog already mentioned introduction of administrative endpoints for channels and subsctriptions. :thinking:

File: src/tasks/admin/delete-subscription.js (hunk 2 above)
Comment (alrik) [reply src/tasks/admin/delete-subscription.js:41]: Let's address this in a dedicated PR

Comment (alrik) [reply src/tasks/admin/delete-subscription.js:41]: https://github.com/realmq/realmq-platform/issues/148

**Pull request merged**